feat(Banner): create Banner component and mount in Page

ahoskins opened the pull request:
This adds a generic `Banner` component. This component will be used in the [licensing private plugin](https://github.com/mesosphere/dcos-ui-plugins-private/pull/636).

This PR also introduces mounting the banner in the existing `Page` component. This will cause the banner to be shown at the top of each page when enabled (when a breach or expiry).

Closes DCOS-19135.

**Testing:**

Paste the following into `componentDidMount()` of `Page.js` component:
```
    const MockBanner = () => {
      return (
        <Banner>
          You have exceeded your 60 day limit by 2 days.
          {" "}
          <a href="https://mesosphere.com/product" className="message-link">
            More Information
          </a>
        </Banner>
      );
    };

    MountService.MountService.registerComponent(MockBanner, "Page:TopBanner");
```

After pasting this, the banner will render:
![screen shot 2017-10-19 at 5 42 10 pm](https://user-images.githubusercontent.com/1527504/31800448-d2e223ca-b4f5-11e7-8c54-ae3ffe3c2bae.png)


**Checklist**
- [ ] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

<!-- More info can be found by clicking the "guidelines for contributing" link above. -->

weblancaster commented on src/js/components/Page.js:
```diff
@@ -179,6 +180,23 @@ var Page = React.createClass({
+      <Banner>
+        You have exceeded your 60 day limit by 2 days.
+        {" "}
+        <a href="https://www.paypal.me/arhoskins" className="message-link">
```
request changes `https://www.paypal.me/weblancaster`

nLight replied:
😂

nLight replied:
We aren't merging it like this are we? :) Suggestion: replace the text and the link with something neutral, that doesn't have licensing traces, i.e. https://mesosphere.com/product/

ahoskins replied:
Lol, I've changed this to mesosphere.com/product for now. We should eventually have a specific docs page we'll direct to, but that doesn't exist yet.

nLight commented on src/js/components/Page.js:
```diff
@@ -179,6 +180,23 @@ var Page = React.createClass({
     );
   },
 
+  getLicensingBanner() {
```
As pointed out in https://github.com/mesosphere/dcos-ui-plugins-private/pull/636#pullrequestreview-70143342 I think we usually try avoiding such words in the open version. Name suggestion `getTopBanner()`

nLight commented on src/js/components/Page.js:
```diff
@@ -179,6 +180,23 @@ var Page = React.createClass({
+  getLicensingBanner() {
+    // todo use mountservice and mount LicenseBanner
+    // which will render the Banner as a child
+    // return <MountService.Mount type="Page:LicenseBanner" />
```
In the respective EE PR it has `DCOS:*` prefix. I think we should come up with a naming scheme for Mounts.

orlandohohmeier replied:
@nLight I'd prefer to use shared types instead of introducing a naming scheme. Anyway, could you please file a foundation issues to revisit the mount service?

nLight replied:
It's there already ;) https://jira.mesosphere.com/browse/DCOS-18779 one of the in the epic https://jira.mesosphere.com/browse/DCOS-18778

ahoskins commented:
I've renamed the mount point to "Page:TopBanner" and the method to `getTopBanner()` to keep it general, as @nLight suggested.

weblancaster commented on src/js/components/Banner.js:
```diff
@@ -0,0 +1,11 @@
+
+const Banner = props => {
+  return (
+    <div className={"banner-container message message-warning flush-bottom"}>
```
Minor but why not only as string `"banner-container message message-warning flush-bottom"`

ahoskins replied:
Yes I noticed this too, I'll change it 👍

nLight commented on src/js/components/Page.js:
```diff
@@ -192,6 +210,7 @@ var Page = React.createClass({
 
     return (
       <div className={classSet}>
+        {this.getTopBanner()}
```
Maybe we should put `<MountService.Mount type="Page:TopBanner" />` right here and get rid of the `getTopBanner` entirely?

nLight commented on src/js/components/Page.js:
```diff
@@ -179,6 +180,23 @@ var Page = React.createClass({
   },
 
+  getTopBanner() {
+    // todo use mountservice and mount LicenseBanner
```
For testing I'd provide a short snippet in the PR's description

```javascript
MountService.registerComponent(
  <Banner>
    You have exceeded your 60 day limit by 2 days.
    {" "}
    <a href="https://mesosphere.com/product" className="message-link">
      More Information
    </a>
  </Banner>,
  "Page:TopBanner"
);
```

This way I could have merged the PR right away.

ahoskins commented:
Thanks @nLight. I followed your suggestions and removed the `getTopBanner()` method, and directly used `MountService.Mount` in page. I've added a note in the PR description about how to test this now.

MatApple commented on src/styles/components/banner/styles.less:
```diff
@@ -0,0 +1,7 @@
+& when (@banner-enabled) {
+
+  .banner-container {
```
Instead of this, wondering why we couldn't wrap the Banner JSX in a `<div className="pod">`? We could potentially avoid all custom CSS.

ahoskins commented:
I've removed the custom CSS for the banner as @MatApple suggested. Now, we are just using the cnvs `message` to style the banner.

MatApple commented:
Looks great! 💯
